tests/generic/test_inputlist.py

```diff
@@ -297,79 +301,162 @@ def rec(m):
         self.assertTrue(rec(self.pl._repr_json_()),
                 "_repr_json_ output not all str")
 
-    # hdf tests
-    def test_to_hdf(self):
+
+    def test_to_hdf_type(self):
+        """
+        Should write correct type information.
+        """
         self.pl.to_hdf(hdf=self.hdf)
         self.assertEqual(self.hdf["input/NAME"],
                          "InputList")
         self.assertEqual(self.hdf["input/OBJECT"],
                          "InputList")
         self.assertEqual(self.hdf["input/TYPE"],
                          "<class 'pyiron_base.generic.inputlist.InputList'>")
-        l = InputList(self.hdf["input/data"])
-        self.assertEqual(self.pl, l)
 
-        pl = InputList(self.pl)
-        pl.to_hdf(hdf=self.hdf)
-        self.assertEqual(self.hdf["NAME"],
-                         "InputList")
-        self.assertEqual(self.hdf["OBJECT"],
-                         "InputList")
-        self.assertEqual(self.hdf["TYPE"],
-                         "<class 'pyiron_base.generic.inputlist.InputList'>")
-        l = InputList(self.hdf["data"])
-        self.assertEqual(pl, l)
+    def test_to_hdf_items(self):
+        """
+        Should write all sublists to HDF groups and simple items to HDF
+        datasets.
+        """
+        self.pl.to_hdf(hdf=self.hdf)
+        for i, (k, v) in enumerate(self.pl.items()):
+            k = "{}__index_{}".format(k if isinstance(k, str) else "", i)
+            if isinstance(v, InputList):
+                self.assertTrue(k in self.hdf["input"].list_groups(),
+                                "Sublist '{}' not a sub group in hdf!".format(k))
+            else:
+                self.assertTrue(k in self.hdf["input"].list_nodes(),
+                                "Item '{}' not a dataset in hdf!".format(k))
+
+    def test_to_hdf_name(self):
+        """
+        Should raise error if clashing names are given.
+        """
+        with self.assertRaises(ValueError,
+                msg="Cannot have names clashing with index mangling."):
+            InputList({'__index_0': 42}).to_hdf(hdf=self.hdf)
 
     def test_to_hdf_group(self):
+        """
```

Comment:
Nit: short docstrings can be one line instead of three according to [PEP257](https://www.python.org/dev/peps/pep-0257/)